tracing-subscriber/src/filter/env/mod.rs

```diff
@@ -242,6 +242,55 @@ impl EnvFilter {
     }
 
     fn from_directives(directives: impl IntoIterator<Item = Directive>) -> Self {
+        use tracing::level_filters::STATIC_MAX_LEVEL;
+        use tracing::Level;
+
+        let directives: Vec<_> = directives.into_iter().collect();
+
+        let disabled: Vec<_> = directives
+            .iter()
+            .filter(|directive| directive.level > STATIC_MAX_LEVEL)
+            .collect();
+
+        if !disabled.is_empty() {
+            eprintln!("warning: some trace filter directives would enable traces that are disabled statically");
```

Comment:
Nit: can you add a comment explaining why `stderr` was chosen? E.g., I'm assuming that the EnvFilter can't access the underlying `MakeWriter`, let alone assume one is present.

Reply:
fwiw, this function already printed to stderr when directives don't parse correctly (borrowed from `env-filter`)